auth/src/main/java/io/grpc/auth/ClientAuthInterceptor.java

```diff
@@ -96,11 +104,29 @@ protected void checkedStart(Listener<RespT> responseListener, Metadata headers)
     };
   }
 
-  private Map<String, List<String>> getRequestMetadata() {
+  private URI serviceUri(Channel channel, MethodDescriptor<?, ?> method) throws StatusException {
+    String authority = channel.authority();
+    if (authority == null) {
+      throw Status.UNAUTHENTICATED.withDescription("Channel has no authority").asException();
+    }
+    final String suffixToStrip = ":443";
+    if (authority.endsWith(suffixToStrip)) {
+      authority = authority.substring(0, authority.length() - suffixToStrip.length());
+    }
+    String path = "/" + MethodDescriptor.extractFullServiceName(method.getFullMethodName());
+    try {
+      return new URI("https", authority, path, null);
```

Comment:
I don't see SCHEME helping anyone, and it is important that it is "https" as otherwise 443 would be inappropriate.

Concerning non-secure channels, based on #874 I believe the plan is to have a commonly-used utility that does the checking before creating the `ClientAuthInterceptor`.